torch/nn/modules/container.py

```diff
@@ -27,21 +27,50 @@ def __init__(self, **kwargs: Any) -> None:
 
 
 class Sequential(Module):
-    r"""A sequential container.
-    Modules will be added to it in the order they are passed in the constructor.
-    Alternatively, an ordered dict of modules can also be passed in.
+    r"""A list of ``Module``s that acts as a ``Module`` itself.
+    A ``Sequential`` is fundamentally a list of ``Module``s, each with a 
+    ``forward()`` method. ``Sequential`` provides a ``forward()`` method
+    of its own, which accepts any input and forwards it to the first 
+    module it stores. It then "chains" outputs to inputs sequentially 
+    for each subsequent module, finally returning the output of the last
+    module. Modules will be added to it in the order they are passed in 
+    the constructor. Alternatively, an ``OrderedDict`` of modules can 
+    be passed in.
+
+    Why should you use a ``Sequential`` instead of a ``ModuleList``? 
+    The value a ``Sequential`` provides over manually calling a sequence
+    of modules is that it allows treating the whole container as a 
+    single module, such that performing a transformation on the 
+    ``Sequential`` applies to each of the modules it stores (which are 
+    each a registered submodule of the ``Sequential``). Also, unlike
+    ``Sequential``,  a ``ModuleList`` does not have a ``forward()`` 
+    method, because it does not define a neural network; in fact, there 
+    is no connection between each of the Modules that it stores. A
+    ``ModuleList`` is exactly what it sounds like--a list for storing
+    ``Module``s! On the other hand, the layers in a ``Sequential`` are 
+    connected in a cascading way.
```

Comment:
While I agree this section on `Sequential` vs. `ModuleList` has value, I think it's a bit too long, as it now dominates the `Sequential` docs.

I'd prefer if this was trimmed down to 1-2 lines succinctly differentiating `Sequential` and `ModuleList` at a high level, with a link to `ModuleList` so users can read more there if they're interested.

Link can be created with the syntax:
```
:class:`torch.nn.ModuleList`
```